### DISCUSSION TITLE
JSON API Documentation

### DISCUSSION THREAD

**Discussion by prosticoco** [smallstep/certificates#549] [Q&A] [answered]
Hello,

Is there any documentation to the JSON API for the `step-ca` server?

**Comment by mmalone** [comment, marked as answer]
Hi! Unfortunately, no. We do not have good documentation on the JSON API. It's something we've had on our docs roadmap for a while, but we don't have an ETA yet.

The closest thing to accessible API docs that we have at the moment are the comments in the `tinystep` Rust library. The library is really well done and the comments describing the API are readable even if you don't know Rust. Check out https://github.com/Bluestone/tinystep/tree/trunk/src.

If you have any specific questions about the API I'm happy to help answer them here.

**Comment by maraino** [comment]
Another way to figure out the API is looking at the Client implementation used by `step` and `step-ca`:
https://github.com/smallstep/certificates/blob/75f24a103adeee22a24e2b5e3bcd664c8d43340c/ca/client.go#L484-L1051
